backend/tournesol/entities/base.py

```diff
@@ -220,12 +217,54 @@ def refresh_metadata(self, *, force=False, save=True):
             return
 
         self.instance.last_metadata_request_at = timezone.now()
-        if save:
-            self.instance.save(update_fields=["last_metadata_request_at"])
 
         self.update_metadata_field()
         # Ensure that the metadata format is valid after refresh
         self.instance.metadata = self.cleaned_metadata
         self.instance.metadata_timestamp = timezone.now()
         if save:
-            self.instance.save(update_fields=["metadata", "metadata_timestamp"])
+            self.instance.save(
+                update_fields=["last_metadata_request_at", "metadata", "metadata_timestamp"]
+            )
```

Comment:
@glerzing What's the motivation for this change?

I think the initial intention was to save `last_metadata_request_at` separately, as a precaution in the case when `self.update_metadata_field()` would fail unexpectedly. For videos metadata especially, it would prevent from requesting Youtube API repeatedly (and potentially abusively) because of a temporary issue (API unavailability, invalid credentials, etc).

Reply:
This was just to avoid doing unnecessary saves.
Did I miss some type of unhandled exception?
Anyway, you can revert this if you deem it safer.

Reply:
I reverted the change, and added a comment to explain why it's safer.